main.py

```diff
@@ -501,89 +513,45 @@ def main(cl_arguments):
-                    task_to_use = model._get_task_params(task.name).get("use_classifier", task.name)
-                    if task.name != task_to_use:
-                        task_model_to_load = task_to_use
-                    else:
-                        task_model_to_load = task.name
-
-                    # Special checkpointing logic here since we train the sentence encoder
-                    # and have a best set of sent encoder model weights per task.
-                    finetune_path = os.path.join(
-                        args.run_dir, "model_state_%s_best.th" % task_model_to_load
-                    )
-                    if os.path.exists(finetune_path):
-                        ckpt_path = finetune_path
-                    else:
-                        # Find the task-specific best checkpoint to evaluate on.
-                        ckpt_path = get_best_checkpoint_path(
-                            args.run_dir, "target_train", task.name
-                        )
-                    assert ".best" in ckpt_path
-                    load_model_state(
-                        model, ckpt_path, args.cuda, skip_task_models=[], strict=strict
-                    )
 
-                    evaluate_and_write(args, model, [task], splits_to_write)
-            elif args.do_pretrain:
-                # If args.do_target_task_training = 0 and args.do_pretrain = 1
-                # then evaluate on pretraining checkpoints.
-                for task in pretrain_tasks:
-                    ckpt_path = get_best_checkpoint_path(args.run_dir, "pretrain", task.name)
-                    assert ".best" in ckpt_path
-                    load_model_state(
-                        model, ckpt_path, args.cuda, skip_task_models=[], strict=strict
+        if args.do_target_task_training:
+            for task in target_tasks:
+                # Find the task-specific best checkpoint to evaluate on.
+                if args.load_eval_checkpoint:
+                    checkpoint = glob.glob(args.load_eval_checkpoint)
+                    assert len(checkpoint) > 0, (
+                        "Specified load_eval_checkpoint not found: %r"
+                        % args.load_eval_checkpoint
                     )
-                    evaluate_and_write(args, model, [task], splits_to_write)
-
-        elif args.transfer_paradigm == "frozen":
-            # Don't do any special checkpointing logic here
-            # since model already has all the trained task specific modules.
-            evaluate_and_write(args, model, target_tasks, splits_to_write)
+                    assert len(checkpoint) == 1, (
+                        "Too many checkpoints match pattern: %r" % args.load_eval_checkpoint
+                    )
+                    ckpt_path = checkpoint[0]
+                else:
+                    ckpt_path = get_best_checkpoint_path(args.run_dir, "target_train", task.name)
+                assert ".best" in ckpt_path
```

Comment:
Might be cleaner to fold this into the get_best_checkpoint_path method, even if it means another keyword arg.